repobee_sanitizer/sanitizer.py

```diff
@@ -77,9 +84,24 @@ def command(self) -> Optional[plug.Result]:
             errors = _sanitize_repo.sanitize_files(repo_root, file_relpaths)
         else:
             LOGGER.info(f"Sanitizing repo and updating {self.target_branch}")
+
+            try:
+                effective_target_branch = (
+                    self._resolve_effective_target_branch(repo_root)
+                )
+            except _sanitize_repo.EmptyTargetBranchError:
```

Comment:
I like using the `_resolve_effective_target_branch` function, but I don't think it makes sense to read when that function is expected to raise an EmptyTargetBranchError

That makes me want to go back to something similar to what we had before:

```python
if self.create_pr_branch:
    try:
        _sanitize_repo.check_empty_target_branch(repo_root, self.target_branch)
    except _sanitize_repo.EmptyTargetBranchError:
        return plug.Result(
            name="sanitize-repo",
            msg=f"Can not create a pull request branch from empty "
            f"target branch {self.target_branch}",
            status=plug.Status.ERROR,
        )

    self.target_branch =  _sanitize_repo.create_pr_branch(
        repo_root, self.target_branch
    )
```

This also means we can remove the helper function `_resolve_effective_target_branch` so the total lines of codes should be shorter, i think this is more intuitive

Reply:
The reason this gets messy is that you're mixing input validation (checking if the target branch exists when running with `--create-pr-branch`) and business logic that relies on that assumption.

Separate these concerns. You can validate the input immediately, at the very top of the `command` method. Just add a method `_validate_input(self)` and perform any input validation you need in there that isn't covered by the CLI itself. For example, that the target branch exists if `--create-pr-branch` is specified.